lib/internal/fs.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const Buffer = require('buffer').Buffer;
+const Stream = require('stream').Stream;
+const fs = require('fs');
+
+function assertEncoding(encoding) {
+  if (encoding && !Buffer.isEncoding(encoding)) {
+    throw new Error('Unknown encoding: ' + encoding);
+  }
+}
+
+// Temporary hack for process.stdout and process.stderr when piped to files.
+class SyncWriteStream extends Stream {
+  constructor(fd, options) {
```

Comment:
`options = {}`? Or have we not checked if default params are speedy enough?

(Quite frankly it doesn't matter here though.)